Library/Dependencies/napi/source/js_native_api_chakra.cc

```diff
@@ -772,164 +464,85 @@ const char* error_messages[] = {
 
 napi_status napi_get_last_error_info(napi_env env,
                                      const napi_extended_error_info** result) {
-  CHECK_ARG(result);
-
-  // $HACK
-  //static_assert(
-  //  node::arraysize(error_messages) == napi_bigint_expected + 1,
-  //  "Count of error messages must match count of error values");
-  //assert(static_last_error.error_code <= napi_callback_scope_mismatch);
+  CHECK_ENV(env);
+  CHECK_ARG(env, result);
+
+  // you must update this assert to reference the last message
+  // in the napi_status enum each time a new error message is added.
+  // We don't have a napi_status_last as this would result in an ABI
+  // change each time a message was added.
+  static_assert(
+    std::size(error_messages) == napi_bigint_expected + 1,
+    "Count of error messages must match count of error values");
+  assert(env->last_error.error_code <= napi_callback_scope_mismatch);
 
   // Wait until someone requests the last error information to fetch the error
   // message string
-  static_last_error.error_message =
-    error_messages[static_last_error.error_code];
+  env->last_error.error_message =
+    error_messages[env->last_error.error_code];
 
-  *result = &static_last_error;
+  *result = &env->last_error;
   return napi_ok;
 }
 
-void napi_clear_last_error() {
-  static_last_error.error_code = napi_ok;
-  static_last_error.engine_error_code = 0;
-  static_last_error.engine_reserved = nullptr;
-}
-
-// TODO(digitalinfinity): Does this need to be on the
-// environment instead of a static error?
-napi_status napi_set_last_error(napi_status error_code,
-                                uint32_t engine_error_code,
-                                void* engine_reserved) {
-  static_last_error.error_code = error_code;
-  static_last_error.engine_error_code = engine_error_code;
-  static_last_error.engine_reserved = engine_reserved;
-
-  return error_code;
-}
-
-napi_status napi_set_last_error(JsErrorCode jsError, void* engine_reserved) {
-  napi_status status;
-  switch (jsError) {
-    case JsNoError: status = napi_ok; break;
-    case JsErrorNullArgument:
-    case JsErrorInvalidArgument: status = napi_invalid_arg; break;
-    case JsErrorPropertyNotString: status = napi_string_expected; break;
-    case JsErrorArgumentNotObject: status = napi_object_expected; break;
-    case JsErrorScriptException:
-    case JsErrorInExceptionState: status = napi_pending_exception; break;
-    default: status = napi_generic_failure; break;
-  }
-
-  static_last_error.error_code = status;
-  static_last_error.engine_error_code = jsError;
-  static_last_error.engine_reserved = engine_reserved;
-  return status;
-}
-
-// $HACK
-//napi_status napi_fatal_exception(napi_env env, napi_value err) {
-//  CHECK_ARG(err);
+// TODO
```

Comment:
Is there an issue to track this?

Reply:
#6 Correctly propagate and handle JavaScript errors

Reply:
I've removed this part. We need to implement napi_fatal_error somewhere but it doesn't need to be JS API specific.